Where is 'State' defined in social_graph_snippets.py?

__State__ is currently an _undefined name_ in the context of social_graph_snippets.py so the code would raise a NameError at runtime.

Is __State__ merely an Enum with two items (__visited__ and __unvisited__) or is it more complex than that?  If it is just the simple Enum then perhaps it would be cleaner to rename the field to be __source.state_visited__ and use the values __True__ and __False__.

flake8 testing of https://github.com/donnemartin/system-design-primer on Python 3.6.3

$ __flake8 . --count --select=E901,E999,F821,F822,F823 --show-source --statistics__
```
./solutions/system_design/social_graph/social_graph_snippets.py:10:30: F821 undefined name 'State'
        source.visit_state = State.visited
                             ^
./solutions/system_design/social_graph/social_graph_snippets.py:17:49: F821 undefined name 'State'
                if adjacent_node.visit_state == State.unvisited:
                                                ^
./solutions/system_design/social_graph/social_graph_snippets.py:19:49: F821 undefined name 'State'
                    adjacent_node.visit_state = State.visited
                                                ^
3    F821 undefined name 'State'
```

Discovered via #93

--- a/solutions/system_design/social_graph/social_graph_snippets.py
+++ b/solutions/system_design/social_graph/social_graph_snippets.py
@@ -1,5 +1,11 @@
 # -*- coding: utf-8 -*-
 from collections import deque
+from enum import Enum
+
+
+class State(Enum):
+    unvisited = 0
+    visited = 1
 
 
 class Graph(object):

